packages/vulcan-core/lib/modules/default_mutations.js

```diff
@@ -82,164 +83,197 @@ export const getDefaultMutations = (collectionName, options = {}) => {
         Utils.performCheck(this.check, context.currentUser, document);
 
         // call editMutator boilerplate function
-        return await editMutator({
-          collection, 
-          documentId: documentId, 
-          set: set, 
-          unset: unset, 
+        return await updateMutator({
+          collection,
+          documentId,
+          data,
           currentUser: context.currentUser,
           validate: true,
           context,
         });
       },
-
-    },
-
+    };
+    mutations.update = updateMutation;
+    // OpenCRUD backwards compatibility
+    mutations.edit = updateMutation;
+  }
+  if (options.upsert) {
     // mutation for upserting a specific document
-    upsert: {
-      name: `${collectionName}Upsert`,
-
-      description: `Mutation for upserting a ${collectionName} document`,
+    mutations.upsert = {
+      description: `Mutation for upserting a ${typeName} document`,
 
-      async mutation(root, { search, set, unset }, context) {
-        const collection = context[collectionName];
+      async mutation(root, { input }, context) {
+        const { selector, data } = input;
+        const collection = context[typeName];
 
         // check if document exists already
-        const existingDocument = await Connectors.get(collection, search, { fields: { _id: 1 } });
+        const existingDocument = await Connectors.get(collection, selector, { fields: { _id: 1 } });
```

Comment:
Does this still return 1 document? We should make sure to query only 1 document.

Reply:
Side note, we should probably make a setting for what the `_id` field is called in each schema. Since we are wanting to be truly database agnostic, eventually we need to start replacing all these default `_id` with `getSetting('idFieldName', '_id');`